Feat: "카테고리 상세 화면 및 프랜차이즈 화면"

mouse0429 opened the pull request:
### 개요
카테고리 탭
Resolves: #55 

### 추가사항
- [X] 카테고리 별 상품 화면
- [X] 프렌차이즈 화면

### 변경사항 및 이유
* CountAndSortTitle 위젯
타이틀 수량 정렬 기능이 있는 위젯이고 좀 더 확장성 있게 사용하기 위해서 기존에 String 으로 받던 정렬 인자를 Order 라는 타입을 따로 뽑아 변경했습니다.

mouse0429 committed:
fix(count_and_sort_title): 정렬 헤더 위젯 확장

- sort 파라미터 String -> Order 로 타입 변경

mouse0429 committed:
fix(non_member_view): 바텀 패딩 에러 해결

mouse0429 committed:
feat(category_view): 카테고리 별 상품 화면

mouse0429 committed:
feat(franchise_card): 프랜차이즈 카드 위젯

mouse0429 committed:
feat(franchise_view): 프랜차이즈 화면

mouse0429 committed:
fix(routes): 카테고리, 프랜차이즈 라우터 추가

wjlee611 commented on lib/constants/routes.dart:
```diff
@@ -3,6 +3,10 @@ class Routes {
   static const String onBoarding = '/on-boarding';
   static const String search = '/search';
 
+  /// ### Extra - Required
+  static const String category = '/category';
+  static const String franchise = '$category/franchise';
+
```
1. category route에 extra 데이터에 무슨 데이터 타입이 들어가는지 명시해주면 좋을 거 같습니다!

```dart
  /// ### Extra - Required
  /// extra: ItemCategory
  static const String category = '/category';
```

2. franchise 에 $category 가 동적으로 할당되는 건가요?
그렇다면 관련 로직이 어떻게 동작(할당)되는지 설명을 적어주시면 좋을 거 같습니다!
아니라면 /franchise 만 남기거나, category의 sub-route로 설정하시는 게 좋을 거 같습니다!

mouse0429 replied:
1. extra 에 들어가는 타입이 ItemCategory 라는 enum 입니다!
2. 초기에 선택된 카테고리를 init 하는 용도입니다! 
의도적으로 sub-route 가 아닌 extra 사용했는데 sub-route 를 계속 바꿔주기엔 /category 화면에서도 카테고리를 선택해 값을 계속 불러와서요

mouse0429 replied:
관련 주석을 추가했습니다! 
ItemCategory 가 추후 ECategory 로 대체될 예정이므로 추후 enum 리팩토링하면서 주석이 한 번 더 변경될 예정입니다!
@wjlee611

mouse0429 committed:
chore(routes): 관련 주석 추가

- routes class 에 새롭게 추가된 값에 대한 설명 주석추가
- CustomRouter 에 불필요하게 들어있던 initialExtra 삭제

mouse0429 committed:
fix(routes): 프랜차이즈 탭 경로 수정

wjlee611 reviewed (approved):
LGTM!
